fv3core/stencils/c_sw.py

```diff
@@ -47,26 +46,46 @@ def absolute_vorticity(vort: FloatField, fC: FloatFieldIJ, rarea_c: FloatFieldIJ
         vort[0, 0, 0] = fC + rarea_c * vort
 
 
-@gtscript.function
-def nonhydro_x_fluxes(delp: FloatField, pt: FloatField, w: FloatField, utc: FloatField):
-    fx1 = delp[-1, 0, 0] if utc > 0.0 else delp
-    fx = pt[-1, 0, 0] if utc > 0.0 else pt
-    fx2 = w[-1, 0, 0] if utc > 0.0 else w
-    fx1 = utc * fx1
-    fx = fx1 * fx
-    fx2 = fx1 * fx2
-    return fx, fx1, fx2
+def fill_corners_x_delp_pt_w(
+    delp: FloatField,
+    pt: FloatField,
+    w: FloatField,
+):
+
+    with computation(PARALLEL), interval(...):
+        delp = corners.fill_corners_2cells_x(delp)
```

Comment:
If you want to reduce code duplication slightly, `corners.fill_corners_2cells_x` can instead be an external `corners_func` and you can compile the same stencil twice: once for the `_x` variant and once for the `_y`.

Reply:
We had that in master previously, I thought there was a reason we removed it.

Reply:
Oh, that was because it was happening at runtime and was causing slowness, but it was not due to using the x and y as functions but rather the runtime recompilation, which this should not face. updated to use your suggestion.